src/router_ics_eui_worker.erl

```diff
@@ -0,0 +1,417 @@
+    lager:warning("rcvd unknown call msg: ~p from: ~p", [_Msg, _From]),
+    {reply, ok, State}.
+
+handle_cast(?RECONCILE_START, #state{conn_backoff = Backoff0, route_id = RouteID} = State) ->
+    case get_euis(State) of
+        {error, _Reason} ->
+            {Delay, Backoff1} = backoff:fail(Backoff0),
+            _ = erlang:send_after(Delay, self(), ?INIT),
+            lager:warning("fail to get_euis ~p, retrying in ~wms", [
+                _Reason, Delay
+            ]),
+            {noreply, State#state{conn_backoff = Backoff1}};
+        {ok, _Stream} ->
+            {_, Backoff2} = backoff:succeed(Backoff0),
+            {noreply, State#state{
+                conn_backoff = Backoff2, route_id = RouteID
+            }}
+    end;
+handle_cast(
+    {?RECONCILE_END, EUIPairs}, #state{conn_backoff = Backoff0, route_id = RouteID} = State
+) ->
+    case get_local_eui_pairs(RouteID) of
+        {error, _Reason} ->
+            {Delay, Backoff1} = backoff:fail(Backoff0),
+            _ = erlang:spawn(fun() ->
+                timer:sleep(Delay),
+                ok = ?MODULE:reconcile()
+            end),
+            lager:warning("fail to get local pairs ~p, retrying in ~wms", [_Reason, Delay]),
+            {noreply, State#state{conn_backoff = Backoff1}};
+        {ok, LocalEUIPairs} ->
+            ToAdd = LocalEUIPairs -- EUIPairs,
+            ToRemove = EUIPairs -- LocalEUIPairs,
+            case update_euis([{remove, ToRemove}, {add, ToAdd}], State) of
+                {error, Reason} ->
+                    {noreply, update_euis_failed(Reason, State)};
+                ok ->
+                    lager:info("reconciling done"),
+                    {noreply, State}
+            end
+    end;
+handle_cast(_Msg, State) ->
+    lager:warning("rcvd unknown cast msg: ~p", [_Msg]),
+    {noreply, State}.
+
+handle_info(?INIT, #state{conn_backoff = Backoff0} = State) ->
+    {Delay, Backoff1} = backoff:fail(Backoff0),
+    case connect(State) of
+        {error, _Reason} ->
+            lager:warning("fail to connect ~p, reconnecting in ~wms", [_Reason, Delay]),
+            _ = erlang:send_after(Delay, self(), ?INIT),
+            {noreply, State#state{conn_backoff = Backoff1}};
+        ok ->
+            case get_route_id(State) of
+                {error, _Reason} ->
+                    _ = erlang:send_after(Delay, self(), ?INIT),
+                    lager:warning("fail to get_route_id ~p, reconnecting in ~wms", [_Reason, Delay]),
+                    {noreply, State#state{conn_backoff = Backoff1}};
+                {ok, RouteID} ->
+                    lager:info("connected"),
+                    {_, Backoff2} = backoff:succeed(Backoff0),
+                    ok = ?MODULE:reconcile(),
+                    {noreply, State#state{
+                        conn_backoff = Backoff2, route_id = RouteID
+                    }}
+            end
+    end;
+handle_info({headers, _StreamID, _Data}, State) ->
```

Comment:
With `router_ics_route_get_euis_handler` will this process be getting any of these messages?

Reply:
the other stream (the update and all) still send their headers and all to the worker. for `router_ics_route_get_euis_handler ` I thought it was easier to accumulate in handler rather than more code in worker for all that.